src/components/Login/LoginContainer.tsx

```diff
@@ -101,86 +106,81 @@ export const InitialisationContainer: FunctionComponent<NavigationProps> = ({
     setShowScanner(s => !s);
   };
 
-  const onLogin = async (qrCode: string): Promise<void> => {
-    if (isLoading) return;
-    setIsLoading(true);
-    try {
-      const { key, endpoint } = decodeQr(qrCode);
-      const authenticated = await authenticate(key, endpoint);
-      if (authenticated) {
-        setAuthKey(key);
-        setEndpoint(endpoint);
-        setIsLoading(false);
-        setShowScanner(false);
-        setProducts(authenticated.policies);
-        navigation.navigate("CollectCustomerDetailsScreen");
-      } else {
-        throw new Error("Authentication key is invalid");
-      }
-    } catch (e) {
-      setShowScanner(false);
-      alert(e.message || e);
-      setIsLoading(false);
-    }
-  };
-
   const onBarCodeScanned = (event: BarCodeScanningResult): void => {
     if (!isLoading && event.data) {
+      const qrCode = event.data;
       onToggleScanner();
-      onLogin(event.data);
+      setIsLoading(true);
+      const { key, endpoint } = decodeQr(qrCode);
+      setCodeKey(key);
+      setEndpointTemp(endpoint);
```

Comment:
Hmm, what's the reason for this being 'temp'

Reply:
Cause we can get the endpoint straight from the QR code, but I feel its better to only set it in the context and the async storage after confirming otp, so the component will store inside the local state as a "temp" storage